File: Leetcode/validParenthesis.py
def isValid(self,s: str) -> bool:
    myStack = []
    dict = {
        "}":"{",
        "]":"[",
        ")":"("
    }
    for i in s:
        if(i == "(" or i == "{" or i == "["):
            myStack.append(i)
        else:
            if(len(myStack) > 0):
                if(myStack[-1] != dict[i]):
                    return False
                myStack.pop()
            else: return False
    return len(myStack) == 0

File: Code Forces/validParenthesis.cpp
#include <bits/stdc++.h>
using namespace std;
bool isValid(string s){
    bool valid = true;
    stack <char> myStack;
    unordered_map <char,char> dict = {
        {')','('},
        {'}','{'},
        {']','['},
    };
    for(int i = 0;i<s.length();i++){
        if(s[i] == '(' || s[i] == '[' || s[i] == '{') myStack.push(s[i]);
        else{
           if(myStack.top() != dict[s[i]] || myStack.empty() ) return false;
            myStack.pop(); 
        }
    }
    return myStack.empty();
}
int main(){
    int x;
    cout<<isValid("[]{}");
    cin>>x;
}
